### ISSUE TITLE
Sindragosa

### ISSUE THREAD

**Issue by ConradBunton** [TrinityCore/TrinityCore#18452]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description: Sindragoga unplayable**

**Current behaviour: He kills everyone when he's coming (Include GMs) and then run to the door behind. At this moment, he's unattackable.**

**Expected behaviour:** (Tell us what should happen instead.)

**Steps to reproduce the problem:** Just kill the 2 wyrms and see...

**Branch(es):** 3.3.5

**TC rev. hash/commit: 9486123afbaedfe33d7e9983d7f06f850f73bac6**  

**TDB version: 335.62**  

**Operating system: Tested on Win server 2008 x64 and Win7 x64**  

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by tkrokli** [comment]
This looks very much like issue [Sindragosa Issue](9275) #9275 which got closed, pointing to [[MovementMaps] Sindragosa](8786) #8786, which is also closed. Many other Sindragosa issues have been closed before, pointing to the very old issue [[General] Evade/Movement Issues](4943) #4943 .

Browse this search result list and check each issue for a match to your report:
https://github.com/TrinityCore/TrinityCore/search?q=Sindragosa&state=open&type=Issues&utf8=✓

Please check these 11 other open issues (2 issues are only [DEV] reference lists) and consider closing this as a duplicate of one of the existing issues regarding Sindragosa.

**Comment by Keader** [comment]
It's not movement issue...
she without target is correct behavior at this point...
issue is: When sindragosa appear (intro) she cast some "frostbolt" in raid, and this "frostbolt" hits everyone, including Game Masters...
Ps: Spell name is not frostbolt, lazy dont let me look what is the real name :P

**Comment by lachtanek** [comment]
It does damage based on target count and frost resistance, weren't you alone?
About GM targetting, that one is probably a bug.

**Comment by ariel-** [comment]
It's because Frost bomb intro uses direct DealDamage functions instead of using spell system, does not take into account GM status of Player

**Comment by lachtanek** [comment]
I don't think it should even target GMs through

**Comment by CDawg** [comment]
I agree with @ariel-. Doing research and found that it's not going through the spell system, so regardless of being a GM - you will take damage as a normal player since there is no filter.
Sindragosa was function correctly a few months ago. I don't **think** it has to do with the mmaps.

**Comment by Keader** [comment]
@ariel- maybe one GM check inside script? https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp#L1095

**Comment by Keader** [comment]
About hitting targets outside Line of Sight... spell has attr: ```SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS```

**Comment by tkrokli** [comment]
To allow a GM account character to play as a "normal" character as long as GM is OFF, I guess.

**Comment by ConradBunton** [comment]
Characters also die in GM mode on.
After Sindragosa's landing, he sometimes run to the stair. But he always becomes as unattackable.
The ice wall doesn't appear on the stairs to lock the area.

**Comment by Keader** [comment]
@ConradBunton She will stuck if you dont have boss requeriments (kill all boss before Sindragosa)
if you are behind of this line:
![wowscrnshot_121216_103327](https://cloud.githubusercontent.com/assets/7120913/21101125/d0f297ca-c05e-11e6-8b03-07b5b6ebe2b5.jpg)
She will pull in you but enter in evade mode (caused by boundary system), so she will decrease 1 heroic attempt.
This is correct behavior, because in retail, players should pull Sindragosa after she down to ground... (Has a ```DoZoneInCombat``` call when she reach the point ```POINT_FROSTWYRM_LAND```

**Comment by Keader** [comment]
Hmm, i can reproduce your evade stuck issue, trying find a way to fix...

**Comment by ConradBunton** [comment]
Anothers bugs on this instance :
No supresser on Valithria
NPC 37127 37132 37133 37134 haven't their casts
NPC 37126 sometimes stuck and stay unattackable

**Comment by Keader** [comment]
1 bug per issue :P

**Comment by Killyana** [comment]
It's a boundary system issue, to reproduce it:
- Connect 2 players
- Kill all boss unitl Sindragosa (you have to finish the Gunship Battle and Valithria also)
- Place the character number one at .go 4361.2392, 2482.7885, 358.438
- Kill trash mobs and the 2 drakes to spawn Sindragosa with the second character
- move quickly to the room in front of Sindragosa to avoid the aggro
- Notice Sindragosa will move to Svalna position where the character number one stays

**Comment by Keader** [comment]
i dont think that is boundary issue
because you can reproduce this issue with npcs without boundary system, just add some condition with return in ```EnterCombat()```

**Comment by Killyana** [comment]
But the boss is not supposed to bypass the boundary area.

**Comment by Keader** [comment]
true, but when this issue occours, script stuck, nothing work anymore...
she stuck in some EnterCombat/EnterEvade loop, dont back from home position, everything looks stucked

**Comment by Killyana** [comment]
it's the anticheat system
do this:
.instance setbossstate 0 3
.instance setbossstate 1 3
.instance setbossstate 2 3
.instance setbossstate 3 3
.instance setbossstate 9 3
.instance setbossstate 10 3
Then to reproduce the issue again and again juste .npc del sindragosa and move where she land to make it respawn.
Notice  the position 4361.2392, 2482.7885, 358.438 is just on to pof the boss (25yards)

**Comment by Keader** [comment]
issue here could be related with waypoints too.
Sindra movement is realy weird and he stuck in some loop (maybe caused by #18594)

**Comment by Keader** [comment]
Test with

```diff

diff --git a/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp b/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
index 13dee24..ba50ab5 100644
--- a/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
+++ b/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
@@ -270,10 +270,14 @@ class boss_sindragosa : public CreatureScript
                     return;
                 }
 
-                BossAI::EnterCombat(victim);
                 DoCast(me, SPELL_FROST_AURA);
                 DoCast(me, SPELL_PERMAEATING_CHILL);
                 Talk(SAY_AGGRO);
+
+                instance->SetBossState(DATA_SINDRAGOSA, IN_PROGRESS);
+                me->SetCombatPulseDelay(5);
+                me->setActive(true);
+                DoZoneInCombat();
             }
 
             void EnterEvadeMode(EvadeReason why) override
```

Looks like BossAI::_EnterCobat() cause this evade/stuck issue.
Maybe caused by double check of CheckRequiredBosses

**Comment by Keader** [comment]
Uhh, this bug only happens to GM character.
Players (gm 0) it works well.

**Comment by Aokromes** [comment]
@Keader so this ticket only afects gms?

**Comment by tkrokli** [comment]
... and if so, did ariel-'s commit https://github.com/TrinityCore/TrinityCore/commit/317efd51ec2289b7e22fdcc5ecfafb7c9b05441b solve that part of the issue?

**Comment by Aokromes** [comment]
If it's gm only issue i will close it on 0,0001s :P

**Comment by Killyana** [comment]
No, there's a real issue about boundary, it's easy to reproduce: https://github.com/TrinityCore/TrinityCore/issues/18452#issuecomment-267477155

This issue happen with https://woehead.way-of-elendil.fr/?npc=37129  once Svalna is killed if Crok is inside the boundary area of Sindragosa he will aggro the boss and it will stack the combat.
Probably despawning Crok some seconds after Svalna death fix the problem.

**Comment by Strazhnik** [comment]
Any idea how may despawning Crok after Svalna death? May be increased the distance to the Crok?

**Comment by ariel-** [comment]
Why isn't the boundary cleared temporarily on air phase, and a Z check is added so crok isn't inside boundaries?

**Comment by Strazhnik** [comment]
I don't know.
Reproduce:
1) All dies (2 or more) in Sindragosa and leave the corpse
2) Sindragosa enters the combat with Captain Brendon

**Comment by RR2739-VCN** [comment]
I believe this issue is still valid, Sindragosa becomes "Invalid Target" and wipes the whole raid in seconds., 90% of the times Sindragosa fight is an instant fail due to "Invalid Target" bug.
